Shared/Samples/Show device location with NMEA data sources/ShowDeviceLocationWithNMEADataSourcesView.swift

```diff
@@ -0,0 +1,128 @@
+// limitations under the License.
+
+import SwiftUI
+import ArcGIS
+import ExternalAccessory
+
+struct ShowDeviceLocationWithNMEADataSourcesView: View {
+    /// The view model for the sample.
+    @StateObject private var model = Model()
+    
+    /// A Boolean value indicating whether to show an alert.
+    @State private var isShowingAlert = false
+    
+    /// An error returned from the `EAAccessoryManager`.
+    @State private var error: AccessoryError? {
+        didSet { isShowingAlert = error != nil }
+    }
+    
+    var body: some View {
+        MapView(map: model.map)
+            .locationDisplay(model.locationDisplay)
+            .overlay(alignment: .top) {
+                VStack(alignment: .leading) {
+                    Text(model.accuracyStatus)
+                    Text(model.satelliteStatus)
+                }
+                .frame(maxWidth: .infinity)
+                .padding(.vertical, 6)
+                .background(.thinMaterial, ignoresSafeAreaEdges: .horizontal)
+            }
+            .toolbar {
+                ToolbarItemGroup(placement: .bottomBar) {
+                    Menu("Source") {
+                        Button("Mock Data") {
+                            model.start(usingMockedData: true)
+                        }
+                        Button("Device") {
+                            selectDevice()
+                        }
+                    }
+                    .disabled(model.isSourceMenuDisabled)
+                    Spacer()
+                    Button("Recenter") {
+                        model.autoPanMode = .recenter
+                    }
+                    .disabled(model.isRecenterButtonDisabled)
+                    Spacer()
+                    Button("Reset") {
+                        model.reset()
+                    }
+                    .disabled(model.isResetButtonDisabled)
+                }
+            }
+            .alert("Error", isPresented: $isShowingAlert, presenting: error) { _ in
+                EmptyView()
+            } message: { error in
+                Text(error.detail)
+            }
+            .onDisappear {
+                // Reset the model to stop the data source and observations.
+                model.reset()
+            }
+    }
+    
+    func selectDevice() {
+        if let (accessory, protocolString) = model.firstSupportedAccessoryWithProtocol() {
+            // Use the supported accessory directly if it's already connected.
+            model.accessoryDidConnect(connectedAccessory: accessory, protocolString: protocolString)
+        } else {
+            error = AccessoryError(
+                detail: "There are no supported Bluetooth devices connected. Open up \"Bluetooth Settings\", connect to your supported device, and try again."
+            )
+        
+            // NOTE: The code below shows how to use the built-in Bluetooth picker
+            // to pair a device. However there are a couple of issues that
```

Comment:
Should the commented out code remain in the sample?

Reply:
I'm uncertain on this. On the one hand, I see there is no point to keep useless code. The slightest chance when someone needs this snippet is that one day Apple solves the mentioned UI bug, and the sample is not running on an Apple Silicon device, which I'd say <5% chance. On the other hand, this is the official way of connecting to an Bluetooth external accessory recommended by Apple, so I feel obliged to leave it in to be a good citizen. What's your thought?

here is some discussion with Mark https://github.com/Esri/arcgis-maps-sdk-swift-samples/pull/144#discussion_r1169099933

Reply:
I'm okay with leaving this as-is then.